src/Microsoft.IIS.WebManager.Middleware/src/IISManagerExtension.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Rewrite;
+
+namespace Microsoft.IIS.WebManager.Middleware

```

Comment:
It seems that this "Middleware" is a bit redundant, since it doesn't really implement Invoke for request processing. Instead, it just configures other middlewares to the pipeline.

You can simply UseRewriter, UseDefaultFiles, and UseStaticFiles in the app's Configure function.

Reply:
The way I think of this function is that this middleware is simply bundling these 3 features are they are required for the site to work. The point of this function is so that any dotnet application can simply using 1 line of code. It's better for maintenance.

Reply:
Context: the plan was that IIS Administration API would just call `app.UseIISWebManager()` to deploy the website.